client/context/query.go

```diff
@@ -304,12 +309,86 @@ func (ctx CLIContext) query(path string, key cmn.HexBytes) (res []byte, err erro
 		return res, errors.Errorf("query failed: (%d) %s", resp.Code, resp.Log)
 	}
 
+	// Data from trusted node or subspace query doesn't need verification
+	if ctx.TrustNode || !isQueryStoreWithProof(path) {
+		return resp.Value, nil
+	}
+
+	err = ctx.verifyProof(path, resp)
+	if err != nil {
+		return nil, err
+	}
+
 	return resp.Value, nil
 }
 
+// verifyProof perform response proof verification
+func (ctx CLIContext) verifyProof(path string, resp abci.ResponseQuery) error {
+
```

Comment:
Indeed, this TODO doesn't make sense. If the gaia-lite works on distrust mode, it must specify proper certifier, or it will encounter errors. So how about just return error here? The error message will remind user to specify proper certifier.

Reply:
Do you have any further suggestions?

Reply:
None here -- just the ones I left on the most recent review :-)